airflow/api_connexion/endpoints/backfill_endpoint.py

```diff
@@ -0,0 +1,181 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import logging
+from functools import wraps
+from typing import TYPE_CHECKING
+
+import pendulum
+from sqlalchemy import select
+
+from airflow.api_connexion import security
+from airflow.api_connexion.exceptions import Conflict, NotFound
+from airflow.api_connexion.schemas.backfill_schema import (
+    BackfillCollection,
+    backfill_collection_schema,
+    backfill_schema,
+)
+from airflow.models.backfill import Backfill
+from airflow.models.serialized_dag import SerializedDagModel
+from airflow.utils import timezone
+from airflow.utils.session import NEW_SESSION, provide_session
+from airflow.www.decorators import action_logging
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+
+    from airflow.api_connexion.types import APIResponse
+
+log = logging.getLogger(__name__)
+
+RESOURCE_EVENT_PREFIX = "dag"
+
+
+def backfill_to_dag(func):
+    """
+    Enrich the request with dag_id.
+
+    :meta private:
+    """
+
+    @wraps(func)
+    def wrapper(*, backfill_id, session, **kwargs):
+        backfill = session.get(Backfill, backfill_id)
+        if not backfill:
+            raise NotFound("Backfill not found")
+        return func(dag_id=backfill.dag_id, backfill_id=backfill_id, session=session, **kwargs)
+
+    return wrapper
+
+
+@provide_session
+def _create_backfill(
+    *,
+    dag_id: str,
+    from_date: str,
+    to_date: str,
+    max_active_runs: int,
+    reverse: bool,
+    dag_run_conf: dict | None,
+    session: Session = NEW_SESSION,
+) -> Backfill:
+    serdag = session.get(SerializedDagModel, dag_id)
+    if not serdag:
+        raise NotFound(f"Could not find dag {dag_id}")
+
+    br = Backfill(
+        dag_id=dag_id,
+        from_date=pendulum.parse(from_date),
+        to_date=pendulum.parse(to_date),
+        max_active_runs=max_active_runs,
+        dag_run_conf=dag_run_conf,
+    )
+    session.add(br)
+    session.commit()
+    return br
+
+
+@security.requires_access_dag("GET")
```

Comment:
Should we create a new `DagAccessEntity` entry for backfills?

Reply:
I dunno!  You tell me.

It's easy enough to add an enum value.  But I looked into it and every one has to be mapped to a fab resource, it seems.

E.g. see here https://github.com/apache/airflow/blob/8741e9c1761931c7cff135d53b589053a04f58c1/airflow/providers/fab/auth_manager/fab_auth_manager.py#L99-L103

Will this still be necessary if we are ripping out fab?  In either case, I'm not planning to pick up the UI part of this for a bit so would be nice to defer this part of it if possible.

Reply:
> Will this still be necessary if we are ripping out fab?

Yes since we are still supporting FAB auth manager in Airflow 3

> In either case, I'm not planning to pick up the UI part of this for a bit so would be nice to defer this part of it if possible

Sure, I can do it as follow-up PR if you want but ideally I think we should add a new enum value and we can associate it to an existing FAB resource so that it is backward compatible

Reply:
i added it; please take a look

Reply:
ok @vincbeck since it seems to introduce a lot of complications (see test failures [here](https://github.com/apache/airflow/pull/42475/files#diff-86edd78c9ae6d7bd14be12d16870d2c900014027e53295d67f0f1ab80169825fR132-R134)) i will remove the dag access entity part for now and create a distinct followup PR for that specifically.  this one can just be about the basic endpoint structure and we can refine the security logic in a PR focused specifically on that.

Reply:
Sounds good. The code looks good to me. Yes please, I think we should definitely create a new `DagAccessEntity` value and btw I think we should also create a new `RESOURCE_DAG_BACKFILL`. On the long run I think this is the best solution. It looks like a lot of work but I dont think it is that much. I am happy to assist/help you on that